qiskit/ignis/verification/tomography/fitters/gateset_fitter.py

```diff
@@ -0,0 +1,662 @@
+        self.initial_Gs_E = initial_Gs_E
+        self.d = np.shape(Gs[0])[0]
+        self.n = len(Gs)
+        self.rho = rho
+
+    def x_to_Gs_E(self, x: np.array) -> List[np.array]:
+        """Converts the gauge to the gateset defined by it
+                Args:
+                    x: An array representation of the B matrix
+
+                Returns:
+                    The gateset obtained from B
+
+                Additional information:
+                    Given a vector representation of B, this functions
+                    produces the list [B*G1*B^-1,...,B*Gn*B^-1]
+                    of gates correpsonding to the gauge B
+
+            """
+        B = np.array(x).reshape((self.d, self.d))
+        try:
+            BB = np.linalg.inv(B)
+            return [BB @ G @ B for G in self.initial_Gs_E]
+        except np.linalg.LinAlgError:
+            return np.inf
+
+    def obj_fn(self, x: np.array) -> float:
+        """The norm-based score function for the gauge optimizer
+            Args:
+                x: An array representation of the B matrix
+
+            Returns:
+                The sum of norm differences between the ideal gateset
+                and the one corresponding to B
+        """
+        Gs_E = self.x_to_Gs_E(x)
+        return sum([np.linalg.norm(G - G_E)
+                    for (G, G_E)
+                    in zip(self.Gs, Gs_E)])
+
+    def optimize(self) -> List[np.array]:
+        """The main optimization method
+            Returns:
+                The optimal gateset found by the gauge optimization
+        """
+        initial_value = np.array([(F @ self.rho).T[0] for F in self.Fs]).T
+        result = opt.minimize(self.obj_fn, initial_value)
+        return self.x_to_Gs_E(result.x)
+
+
+def split_list(l: List, sizes: List) -> List[List]:
+    """Splits a list to several lists of given size
+        Args:
+            l: A list
+            sizes: The sizes of the splitted lists
+        Returns:
+            The splitted lists
+        Example:
+            >> split_list([1,2,3,4,5,6,7], [1,4,2])
+            [[1],[2,3,4,5],[6,7]]
+    """
+    if sum(sizes) != len(l):
+        msg = "Length of list ({}) " \
+              "differs from sum of split sizes ({})".format(len(l), sizes)
+        raise RuntimeError(msg)
+    result = []
+    i = 0
+    for s in sizes:
+        result.append(l[i:i + s])
+        i = i + s
+    return result
+
+
+def matrix_to_vec(A: np.array) -> np.array:
+    """Converts a complex matrix to vector representation
+        Args:
+            A: A nxn complex matrix
+        Returns:
+            A row-stacking vector representation of A; first the real part
+            of A and then the imaginary part
+    """
+    real = []
+    imag = []
+    for row in A:
+        for x in row:
+            real.append(np.real(x))
+            imag.append(np.imag(x))
+    return real + imag
+
+
+def vec_to_complex_matrix(vec: np.array, n: int) -> np.array:
```

Comment:
Is function is a little confusing.

If it converting a 2 * n^2 real vector to a n x n complex matrix? If so the shape and type of the vector is real should be added to the doc string.

Also is it just doing standard row-major de-vectorization? If so it can be done much more efficiently using Numpy functions. Eg:
```python
size = n * n
return np.reshape(vec[0:size] + 1j * vec[size: 2 * size], (n, n))
```

Reply:
The second. Not sure why I didn't use np here; I guess I tried, something went wrong and I didn't want to debug it at the time.